### PR TITLE
wfnews-1329 background runner for offline RoF submission

### PR DESCRIPTION
**Pull request by ssylver93**

### CODE REVIEW

File: client/wfnews-war/src/main/angular/src/background.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+          // Remove the locally stored data if sync is successful
+          await storage.remove('offlineReportData');
+        }
+      }
+    } catch (error) {
+      console.error('Sync failed:', error);
+    }
+  }
+
+async function submitReportToServer(offlineReport){
```
Comment (dhlevi) [inline client/wfnews-war/src/main/angular/src/background.js:43]: These functions exist elsewhere, might be good to keep this in a service class so we dont need to make identical changes in multiple places

File: client/wfnews-war/src/main/angular/src/background.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    const rofJson = JSON.parse(offlineReport);
+    const resource = rofJson.resource;
+    const image1 = rofJson.image1;
+    const image2 = rofJson.image2;
+    const image3 = rofJson.image3;
+    const rofUrl = "https://wfone-notifications-api.dev.bcwildfireservices.com/rof"
```
Comment (dhlevi) [inline client/wfnews-war/src/main/angular/src/background.js:49]: No hard-coded URL's, use an environment specific parameter. This should be in our secrets and applied on build, fetched from the app properties

**Review by dhlevi — CHANGES_REQUESTED** [review]
On the right track. Couple of small things

File: client/wfnews-war/src/main/angular/src/background.js (hunk 1 above)
Comment (dhlevi) [reply client/wfnews-war/src/main/angular/src/background.js:43]: Note: I understand they're in the other PR, so we can likely make that change there, merge it in, then update this one to use the shared code

**Comment by dhlevi** [comment]
check the conflict on package-lock. May need to regenerate or drop it an re PR that bit after a merge

File: client/wfnews-war/src/main/angular/src/app/background/background.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    }
+  }
+
+async function submitReportToServer(offlineReport){
+    const rofJson = JSON.parse(offlineReport);
+    const resource = rofJson.resource;
+    const image1 = rofJson.image1;
+    const image2 = rofJson.image2;
+    const image3 = rofJson.image3;
+    const rofUrl = "https://wfone-notifications-api.dev.bcwildfireservices.com/rof"
```
Comment (dhlevi) [inline client/wfnews-war/src/main/angular/src/app/background/background.js:48]: This needs to be parameterized. No hard-coded URLs

**Comment by ssylver93** [comment]
Replaced by PR #1148

**Pull request closed without merging**